src/main/java/com/sforce/async/BulkConnection.java

```diff
@@ -137,21 +138,21 @@ private JobInfo createOrUpdateJob(JobInfo job, String endpoint) throws AsyncApiE
     }
 
     private JobInfo createOrUpdateJob(JobInfo job, String endpoint, ContentType contentType) throws AsyncApiException {
+        OutputStream out = null;
+        InputStream in = null;
         try {
```

Comment:
nit. You could use https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html for input/output stream, then you won't need to close it explicitly.